src/providers/db2/qgsdb2provider.h

```diff
@@ -0,0 +1,180 @@
+/***************************************************************************
+  qgsdb2provider.h - Data provider for DB2 server
+  --------------------------------------
+  Date      : 2016-01-27
+  Copyright : (C) 2016 by David Adler
+                          Shirley Xiao, David Nguyen
+  Email     : dadler at adtechgeospatial.com
+              xshirley2012 at yahoo.com, davidng0123 at gmail.com
+****************************************************************************
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ ***************************************************************************/
+
+#ifndef QGSDB2PROVIDER_H
+#define QGSDB2PROVIDER_H
+
+#include <qgsvectordataprovider.h>
+#include <qgscoordinatereferencesystem.h>
+#include "qgsgeometry.h"
+#include <QtSql>
+
+/**
+ * @class QgsDb2Provider
+ * @brief Data provider for DB2 server.
+ */
+class QgsDb2Provider : public QgsVectorDataProvider
+{
+    Q_OBJECT
+
+  public:
+    explicit QgsDb2Provider( QString uri = QString() );
+
+    virtual ~QgsDb2Provider();
+
+    /**
+     * Returns a QSqlDatabase object that can connect to DB2 for LUW or z/OS.
+     *
+     * If service is provided, then username and password is required.
+     * If service is not provided, the remaining arguments are required.
+     *
+     * @param connInfo A string containing all connection information.
+     */
+    static QSqlDatabase GetDatabase( const QString &connInfo, QString &errMsg );
+
+    static bool OpenDatabase( QSqlDatabase db );
```

Comment:
GetDatabase -> getDatabase, OpenDatabase -> openDatabase (QGIS standard is camel case)